java-checks/src/test/java/org/sonar/java/filters/FilterVerifier.java

```diff
@@ -143,4 +150,23 @@ public void visitTrivia(SyntaxTrivia syntaxTrivia) {
     }
   }
 
+  private static SonarComponents sonarComponents(InputFile inputFile) {
+    SensorContextTester context = SensorContextTester.create(new File("")).setRuntime(SonarRuntimeImpl.forSonarLint(Version.create(6, 7)));
+    context.setSettings(new MapSettings().setProperty("sonar.java.failOnException", true));
+    SonarComponents sonarComponents = new SonarComponents(null, context.fileSystem(), null, null, null) {
+      @Override
+      public boolean reportAnalysisError(RecognitionException re, InputFile inputFile) {
+        return false;
+      }
+
+      @Override
+      public void addAnalysisError(AnalysisError analysisError) {
+        Fail.fail(String.format("Should not fail analysis (%s)", analysisError.getKind().name()));
```

Comment:
Maybe there is something I'm missing but I feel like there should be a test case triggering this. Currently I can just comment this change and all tests are passing anyway. Even when leaving out your fix on the other commit, I've been unable to reach this (I'm pretty sure I'm doing something wrong here - but I still think we should have an easier way to make sure this does its intended job).

Reply:
Added a test case.